diff --git a/node_modules/@graphql-tools/code-file-loader/cjs/index.js b/node_modules/@graphql-tools/code-file-loader/cjs/index.js
index d7f278b..c074009 100644
--- a/node_modules/@graphql-tools/code-file-loader/cjs/index.js
+++ b/node_modules/@graphql-tools/code-file-loader/cjs/index.js
@@ -24,6 +24,7 @@ const FILE_EXTENSIONS = [
     '.jsx',
     '.vue',
     '.svelte',
+    '.astro',
 ];
 function createGlobbyOptions(options) {
     return { absolute: true, ...options, ignore: [] };
diff --git a/node_modules/@graphql-tools/code-file-loader/esm/index.js b/node_modules/@graphql-tools/code-file-loader/esm/index.js
index e1ec348..624811c 100644
--- a/node_modules/@graphql-tools/code-file-loader/esm/index.js
+++ b/node_modules/@graphql-tools/code-file-loader/esm/index.js
@@ -20,6 +20,7 @@ const FILE_EXTENSIONS = [
     '.jsx',
     '.vue',
     '.svelte',
+    '.astro',
 ];
 function createGlobbyOptions(options) {
     return { absolute: true, ...options, ignore: [] };
